### PR TITLE
Implement provider generation for Golang

### PR DESCRIPTION
**Pull request by GTLiSunnyi**

### CODE REVIEW

**Comment by GTLiSunnyi** [comment]
Implement provider generation for Golang

**Comment by GTLiSunnyi** [comment]
> _No description provided._

File: templates/provider/go/common/config.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	DefaultConfigFileName = "config.yaml"
```
Comment (chengwenxi) [inline templates/provider/go/common/config.go]: unused const

File: templates/provider/go/common/config.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+var (
+	ConfigPath = os.ExpandEnv("$HOME/.{{service_name}}-sp/config.yaml")
```
Comment (chengwenxi) [inline templates/provider/go/common/config.go]: get user home via
```go
userHomeDir, err := os.UserHomeDir()
```

File: templates/provider/go/service/config.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/spf13/viper"
+	"github.com/irisnet/service-gen/common"
+)
+
+// default config variables
+var (
+	defaultChainID       = "irita-hub"
+	defaultNodeRPCAddr   = "http://127.0.0.1:26657"
+	defaultNodeGRPCAddr  = "127.0.0.1:9090"
+	defaultKeyPath       = os.ExpandEnv(filepath.Join("$HOME", ".iritacli"))
```
Comment (chengwenxi) [inline templates/provider/go/service/config.go]: ditto.

File: templates/provider/go/common/config.go [hunk 4]
```diff
@@ -2,17 +2,13 @@ package common
...
 	"github.com/spf13/viper"
 )
 
-const (
-	DefaultConfigFileName = "config.yaml"
-)
-
 var (
-	ConfigPath = os.ExpandEnv("$HOME/.{{service_name}}-sp/config.yaml")
+	ConfigPath = utils.MustGetHomeDir()
```
Comment (chengwenxi) [inline templates/provider/go/common/config.go]: utils.MustGetHomeDir() + ".{{service_name}}-sp/config.yaml"

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**